compiler/rustc_borrowck/src/diagnostics/conflict_errors.rs

```diff
@@ -446,10 +450,69 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
                 }
             }
         }
+
         err.span_label(decl_span, "binding declared here but left uninitialized");
+        if show_assign_sugg {
+            self.suggest_assign_rvalue(&mut err, moved_place, &name, decl_span);
+        }
         err
     }
 
+    fn suggest_assign_rvalue(
+        &self,
+        err: &mut Diagnostic,
+        moved_place: PlaceRef<'tcx>,
+        name: &str,
+        decl_span: Span,
+    ) {
+        let ty = moved_place.ty(self.body, self.infcx.tcx).ty;
+        debug!("ty: {:?}, kind: {:?}", ty, ty.kind());
+
+        let initilize_msg = match ty.kind() {
+            ty::Array(_, n) => format!("[val; {}]", n),
+            ty::Int(_) | ty::Uint(_) => format!("0"),
+            ty::Float(_) => format!("0.0"),
+            ty::Bool => format!("false"),
+            ty::Never | ty::Error(_) => "".to_string(),
+            ty::Adt(def, _substs) => {
+                if format!("{:?}", def).starts_with("std::vec::Vec") {
+                    format!("vec![]")
+                } else if let Some(default_trait) = self.infcx.tcx.get_diagnostic_item(sym::Default) &&
+                    self.infcx.tcx.infer_ctxt().enter(|infcx| {
+                        infcx.type_implements_trait(default_trait, ty, ty::List::empty(), self.param_env).may_apply()
+                    }) {
+                    format!("Default::default()")
+                } else {
+                    format!("something")
+                }
+            },
+            _ => format!("something"),
+        };
+
+        if initilize_msg.is_empty() {
+            return;
+        }
+
+        let sugg_span = self
+            .infcx
+            .tcx
+            .sess
+            .source_map()
+            .span_extend_while(decl_span, |c| c != '\n')
+            .unwrap_or(decl_span);
+        let mut prefix = self.infcx.tcx.sess.source_map().span_to_snippet(sugg_span).unwrap();
```

Comment:
This is a Hack, any better solution?

Reply:
does using `decl_span.shrink_to_hi()` work?

Reply:
For `let name : i32;`, decl_span will contain `name`, 
so `shrink_to_hi()` will move to the position before ':'.

Reply:
ah, it doesn't contain the type? :/ that's unfortunate, then i guess we should somehow get the span containing the type to here. lazily extending that span doesn't seem great to me

Reply:
yes, I'm trying to find the something like parent span....